### DISCUSSION TITLE
I can find window.__TAURI_INTERNALS__.invoke but not window.__TAURI__.invoke

### DISCUSSION THREAD

**Discussion by mimhof1** [tauri-apps/tauri#11586] [Q&A] [answered]
I can find window.__TAURI_INTERNALS__.invoke but not window.__TAURI__.invoke  I feel liek I need to keep going to get window.__TAURI__.invoke available but hitting roadblocks. 

Probably me and not a bug, but I've been editing tauri.conf.json but no success. Trying allowlist and getting unexpected error when compiling.  Any path to a solution I should try? I've also cargo update and messed with toml fro a while.  Please let me know what configs to dump here to see if you can guide me.

**Comment by FabianLars** [comment]
The output of the `tauri info` command as a start would be nice.

The only thing that should be required for this is to enable `build.withGlobalTauri` in v1 or `app.withGlobalTauri` in v2 in tauri.conf.json.

In v2 the `window.__TAURI__.invoke` re-export may not exist anymore, try `window.__TAURI__.core.invoke` instead.

**Comment by mimhof1** [comment]
Thanks Fabian:
cargo tauri info
![image](https://github.com/user-attachments/assets/8cc39ce9-767b-473d-b6a7-d74cd6bade9e)

tauri.conf.json
![image](https://github.com/user-attachments/assets/2bb926fe-b205-4fc1-9958-407349793977)

Cargo.toml
![image](https://github.com/user-attachments/assets/e34f1884-6358-4dbc-aac2-a35c2c16bc78)

app.js
![image](https://github.com/user-attachments/assets/a86de47e-44d8-49c0-aa38-c6e4e0b73919)

index.html
![image](https://github.com/user-attachments/assets/f157384f-1fa9-4020-b8be-f6066b8d726d)

Run time example, showing loaded data using window.__TAURI_INTERNALS, and below in console can see that window.__TAUTI__ is not defined, so sure then that window.__TAURI__.core.invoke is likely not defined.
![image](https://github.com/user-attachments/assets/1fb8a910-5934-4d4c-a94b-4cc0d8d2a52a)

**Details**
My goal for this first attempt using tauri and rust was just to get the environment setup and show I can get the web tools I am used to oriented to the back end core and retrieve data, bring it back. I succeeded, but cheated with window.__TAURI_INTERNALS.  It feels like I have maybe not properly setup a dependency or mixing versions and capabilities because this is my first attempt at learning.

Any and all guidance is really appreciated.

**Reply by FabianLars** [reply, marked as answer]
Looking at your tauri.conf.json file it's indeed this part of my message:
> The only thing that should be required for this is to enable build.withGlobalTauri in v1 or app.withGlobalTauri in v2 in tauri.conf.json.

Since you're using v2, the app block should look like this
```json
{
  "app": {
    "withGlobalTauri": true,
    "windows": ["keep what you had, i'm just too lazy to type"],
    "security": {
      "csp": null
    }
  }
}
```

**Reply by mimhof1** [reply]
Fabian, thank you, I made the change to app.withGlobalTauri and JS fetch call with window.__TAURI__.core.invoke just as you explained.
I also can now see the docs at https://v2.tauri.app/reference/config/#appconfig and the explanation, so your help also assisted in getting me oriented better:
![image](https://github.com/user-attachments/assets/4e168fd5-5ad9-49fb-bfb8-448b5734f66f)
